content/blog/2021/containerqueries.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Getting Started with Container Queries
+sub: Now is the time to begin experimenting with a long requested layout tool. 
+author: davidh
+date: 2021-04-01
+image:
+  src: projects/w3c.jpg
+  alt: W3C logo
+tags:
+  - CSS
+  - CSSWG
+  - Article
+  - Container Queries
+summary: |
+  An early prototype version of **container queries** is now available in [Chrome Canary](https://www.google.com/chrome/canary). What are container queries exactly and how will they change the way we build layouts?
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+ The `@container` query, that elusive feature developers have been requesting and proposing for years, has finally made its debut in browsers. Well, sort of. As of the latest version of [Chrome Canary](https://www.google.com/chrome/canary), the most recent `@container` query proprosal, developed by OddBird's own [Miriam Suzanne](https://www.oddbird.net/authors/miriam/), is available for use behind an experimental flag. What _are_ container queries exactly? And how do they work?
+
+ Bookmark Miriam's scratch site for updates: [Miriam's CSS Sandbox](https://css.oddbird.net/rwd/query/). 
+
+ ## Background
+
+`@Media` queries sparked a responsive design revolution by allowing authors to change the styles of elements based on the size of the entire viewport. Up until now, what could not be done was changing the style of an element based on the size of its parent element. 
+
+## Syntax and an example
+
+ The proposed `@container` query syntax feels a lot like writing a `@media` query, but a main difference is that a `@container` query has to be implemented in two parts.
+
+In this demo (which only works in the latest version of [Chrome Canary](https://www.google.com/chrome/canary)), a `blockquote` is styled differently depending on the size of its container. 
+
+{{ embed.codepen(
+  id='YzpywrZ',
+  title='Container Queries with Block Quotes',
+  user='dvdherron',
+  height=600
+) }}
+
+### Define a containment context
+
+To achieve this, first a containment context must be defined. This lets the browser know which container to query against later. 
+
+```scss
+article,
+section {
+  contain: layout inline-size;
+}
+```
+
+The `contain` property is part of the existing [CSS Containment Module](https://drafts.csswg.org/css-contain/). The `layout` value turns on [layout contaiment](https://drafts.csswg.org/css-contain/#valdef-contain-layout) on the container, which ensures that "nothing outside can affect its internal layout, and vice versa." The `inline-size` value is a proposed change to the Containment Module that would let authors explicitly declare in which dimension containment should be applied. 
```

Comment:
Give a similar one-sentence description of size containment as well?

Reply:
Mentioned size and cleaned up the `layout` sentence